akvo/rsr/static/scripts-src/my-results/utils.js

```diff
@@ -286,91 +286,67 @@ export function fullUpdateVisibility(update, activeFilter) {
+                break;
+            }
+        case c.FILTER_SHOW_APPROVED:
+            {
+                visible = [c.UPDATE_STATUS_APPROVED];
+                break;
+            }
+        default:
+            {
+                // full visibility to all updates if no filter is active
+                return true;
+            }
     }
     return visible.indexOf(update.status) > -1;
 }
 
-function tree(model, parentId) {
-    // Construct a tree representation of the subtree of data with object model[parentId] as root
-    //TODO: refactor, we shouldn't need findChildren here
-    const ids = findChildren(parentId, model).ids;
-    const childModel = childModelName(model);
-    const children = ids.map(cId => {
-        return tree(childModel, cId);
-    });
-    if (children.length > 0) {
-        return { id: parentId, model: model, children: children };
-    } else {
-        return { id: parentId };
-    }
-}
-
 export function flatten(arr) {
     // Flatten an array of arrays
     return arr.reduce((acc, val) => acc.concat(Array.isArray(val) ? flatten(val) : val), []);
 }
 
-function keysList(node, open, MEManagerKeys) {
-    // "Disassemble" the tree representation of the data from tree() and return a list of objects.
-    // Each object holds the collapseID of the corresponding Collapse and its activeKey
-    const key = {
-        collapseId: collapseId(node.model, node.id),
-        activeKey: open
-            ? idsToActiveKey(node.children.map(child => child.id.toString()))
-            : MEManagerKeys || []
-    };
-    const children = node.children.filter(child => child.model !== undefined);
-    const childKeys = children.map(node => keysList(node, open));
-    return flatten([key].concat(childKeys));
-}
-
-export function toggleTree(model, id, open, MEManagerKeys) {
-    const fullTree = tree(model, id);
-    return keysList(fullTree, open, MEManagerKeys);
+export function toggleTree(open, MEManagerKeys) {
+    const modelsList = c.RESULTS_MODELS_LIST;
+    const models = modelsList.slice(1);
+    const fullTree = models.map((model) => {
+        const parentModel = parentModelName(model);
+        const ids = store.getState().models[parentModel].ids;
+        const childKeys = ids.filter(
+            id => store.getState().models[parentModel].objects[id]._meta.children.ids.length > 0
+        ).map((id) => {
+            const keys = store.getState().models[parentModel].objects[id]._meta.children.ids;
+            return { activeKey: open ? idsToActiveKey(keys) : [], collapseId: collapseId(model, id) };
+        });
+        return childKeys;
+    });
+    return flatten(fullTree);
 }
 
 export function openResults(activeKey, isMEManager) {
-    if (isMEManager) {
+    if(isMEManager) {
         return isMEManagerDefaultKeys(activeKey);
     } else {
         return activeKey == undefined || activeKey.length == 0;
     }
 }
 
-export function createToggleKeys(parentId, model, activeKey) {
```

Comment:
If you search for createToggleKeys in the code you'll find that it's present in a number of toggleAll() methods. However all of those methods are dead code, they're not called from anywhere so they can be removed too! Also make sure to remove createToggleKeys from the import statements and toggleAll from the constructors.